testsuite/tests/match-side-effects/partiality.ml

```diff
@@ -44,24 +44,45 @@ val f : t -> int = <fun>
    inside a mutable position. *)
 type t = {a: bool; mutable b: int option}
 
+let simple x =
+  match x with
+  | {b = None} -> 1
+  | {b = Some y} -> y
+;;
+(* Performance expectation: there should not be a Match_failure case. *)
+[%%expect {|
+0
+type t = { a : bool; mutable b : int option; }
+(let
+  (simple/291 =
+     (function x/293 : int
+       (let (*match*/296 =o (field_mut 1 x/293))
+         (if *match*/296 (field_imm 0 *match*/296) 1))))
+  (apply (field_mut 1 (global Toploop!)) "simple" simple/291))
+val simple : t -> int = <fun>
+|}]
+
+(* This more complex case has the switch on [b] split across two cases
+   on [a], so it may need a [Match_failure] for soundness -- it does
+   if the two accesses to [b] are done on different reads of the same
+   mutable field.
+
+   PASS: a single read of [field_mut 1 x], no Match_failure case. *)
 let f x =
   match x with
   | {a = false; b = _} -> 0
   | {a = _;     b = None} -> 1
   | {a = true;  b = Some y} -> y
 ;;
-(* Performance expectation: there should not be a Match_failure case. *)
 [%%expect {|
-0
-type t = { a : bool; mutable b : int option; }
 (let
-  (f/291 =
-     (function x/292 : int
-       (if (field_int 0 x/292)
-         (let (*match*/296 =o (field_mut 1 x/292))
-           (if *match*/296 (field_imm 0 *match*/296) 1))
+  (f/297 =
+     (function x/298 : int
```

Comment:
Mild suggestion: Did you consider adding `-dno-unique-ids` to the test's flags? That would make this part of the test somewhat less sensitive to the code that comes before, for future changes.

Reply:
I don't want to use `-dno-unique-ids` for pattern-matching tests due to the frequent name conflicts with `*match*/nnn` identifiers generated during pattern-matching: if we passed the option, some of the -dlambda outputs would become ambiguous. In this testfile for example, the tests immediately below the one you are commenting binds both `*match*/307` and `*match*/309`.

It would be nice to avoid this by tweaking the pattern-matching compiler to generate distinct names in a deterministic way (my idea is to use the path from the root to the current position as a distinguishing suffix), but I never did the work of implement this (not immediate) quality-of-life improvement.